safe-svg.php

```diff
@@ -728,6 +717,37 @@ protected function str_ends_with( $haystack, $needle ) {
 			return 0 === substr_compare( $haystack, $needle, -$len, $len );
 		}
 
+		/**
+		 * Set custom width or height of the SVG image.
+		 *
+		 * @param int          $id   Image attachment ID.
+		 * @param string|array $size Size of image. Image size or array of width and height values
+		 *                                    (in that order). Default 'thumbnail'.
+		 *
+		 * @return int|false Width or height of the SVG image, or false if not found.
+		 */
+		protected function set_svg_dimensions( $id, $size ) {
+			$dimensions = $this->svg_dimensions( $id );
+
+			if ( is_array( $size ) ) {
+				$width  = $size[0];
+				$height = $size[1];
+			} elseif ( 'full' === $size && $dimensions ) {
```

Comment:
Because the output of `svg_dimensions` can be filtered, I think we'll need to be more defensive in a few places here to ensure 1 - `$dimensions` is an array and 2 - it contains the `height` and `width` keys